### ISSUE TITLE
Some spike elapsed time of ANN benchmark

### ISSUE THREAD

**Issue by zzong2006** [weaviate/weaviate-python-client#223]
I try some (personal) benchmark on distributed HNSW index. 
Deploy 5 pods in k8s and each pod has 2gb RAM / 10 CPU cores.

During benchmark, I noticed some spike on elapsed time on query but cannot figure out.

Here is some configuration / scripts in bench

```python
DATA_SIZE = 2000
QUERY_SIZE = 100
DIM = 256
TOP_K = 100
client = weaviate.Client("http://localhost:8080") 
samples = np.random.randn(DATA_SIZE, DIM).astype("float32")
....

for k in range(QUERY_SIZE):
    start_time = time.time()
    target_query = {"vector": samples[k]}

    result = (client.query.get(TARGET_CLASS, ["index_id"]).with_near_vector(target_query).with_limit(TOP_K).do())
    end_time = time.time()
    elapsed = 1000. * (end_time - start_time)
    print(f"{k}: Elapsed", elapsed)
```

_trial (1)_
```bash
....
91: Elapsed 8.621692657470703
92: Elapsed 8.2550048828125
93: Elapsed 7.802248001098633
94: Elapsed 86.1060619354248
95: Elapsed 7.91478157043457
96: Elapsed 81.93159103393555
97: Elapsed 78.98426055908203  # why this happened?
....
```

_trial (2)_
```bash
91: Elapsed 9.263992309570312
92: Elapsed 87.82005310058594
93: Elapsed 8.857011795043945
94: Elapsed 88.12785148620605
95: Elapsed 9.41157341003418
96: Elapsed 87.73326873779297
97: Elapsed 8.994817733764648  # same vector as previous trial but doesn't show spike this time.
```

**schema**

```
scahem.get {
    "class": "RandomVectors",
    "invertedIndexConfig": {
        "bm25": {
            "b": 0.75,
            "k1": 1.2
        },
        "cleanupIntervalSeconds": 60,
        "stopwords": {
            "additions": null,
            "preset": "en",
            "removals": null
        }
    },
    "properties": [
        {
            "dataType": [
                "int"
            ],
            "indexInverted": false,
            "name": "index_id"
        }
    ],
    "replicationConfig": {
        "factor": 1
    },
    "shardingConfig": {
        "virtualPerPhysical": 128,
        "desiredCount": 1,
        "actualCount": 1,
        "desiredVirtualCount": 128,
        "actualVirtualCount": 128,
        "key": "_id",
        "strategy": "hash",
        "function": "murmur3"
    },
    "vectorIndexConfig": {
        "skip": false,
        "cleanupIntervalSeconds": 300,
        "maxConnections": 48,
        "efConstruction": 128,
        "ef": -1,
        "dynamicEfMin": 100,
        "dynamicEfMax": 500,
        "dynamicEfFactor": 8,
        "vectorCacheMaxObjects": 1000000000000,
        "flatSearchCutoff": 0,
        "distance": "dot"
    },
    "vectorIndexType": "hnsw",
    "vectorizer": "none"
}
```

I changed shardingConfig from 1 to 5 but spike always happened.

**Comment by dirkkul** [comment]
@zzong2006 
Thanks for bringing this up. Does this only happen on distributed setups?

**Comment by zzong2006** [comment]
@dirkkul Well.. I guess not.

Whether I deployed single or multiple(>5) pods, I experienced same issue. 
But I didn't try on docker setup.

**Comment by dirkkul** [comment]
Finally had time to look into this. 

I noticed two things:
1) The times do not seem to depend on the vector that is send. Eg if you send exactly the same queries several times, the times are completely different
2) Almost all the time is spend in the request library that is outside of our control ([Profiling results](https://user-images.githubusercontent.com/5353192/227934293-f36e42a6-2a45-4897-8d5b-89493e65b6d0.svg)). I assume that that library soetimes just takes longer :(

Sadly, I don't think that theres anything we can do here. Parallelizing requests is on the "we should do that at some point"-list, but I' not sure when we have time for that.

Closing for now.
